arangod/IResearch/IResearchLinkCoordinator.cpp

```diff
@@ -83,6 +84,61 @@ IResearchLinkCoordinator::IResearchLinkCoordinator(
   _sparse = true;   // always sparse
 }
 
+Result IResearchLinkCoordinator::init(
+    velocypack::Slice definition,
+    IResearchLink::InitCallback const& initCallback) {
+  auto r = IResearchLink::init(definition, initCallback);
+  if (!r.ok()) {
+    return r;
+  }
+  using namespace metrics;
+  auto& metric = IResearchLink::collection()
+                     .vocbase()
+                     .server()
+                     .getFeature<ClusterMetricsFeature>();
+
+  auto batchToCoordinator = [](ClusterMetricsFeature::Metrics& metrics,
+                               std::string_view name, velocypack::Slice labels,
+                               velocypack::Slice value) {
+    ClusterMetricsFeature::MetricKey key{name, labels.stringView()};
+    std::get<uint64_t>(metrics[key]) += value.getNumber<uint64_t>();
+  };
+  auto batchToPrometheus = [](std::string& result, std::string_view globals,
+                              ClusterMetricsFeature::MetricKey const& key,
+                              ClusterMetricsFeature::MetricValue const& value) {
+    Metric::addMark(result, key.first, globals, key.second);
+    result.append(std::to_string(std::get<uint64_t>(value))) += '\n';
+  };
+  metric.add("arangodb_search_num_docs", batchToCoordinator, batchToPrometheus);
+  metric.add("arangodb_search_num_live_docs", batchToCoordinator,
+             batchToPrometheus);
+  metric.add("arangodb_search_num_segments", batchToCoordinator,
+             batchToPrometheus);
+  metric.add("arangodb_search_num_files", batchToCoordinator,
+             batchToPrometheus);
+  metric.add("arangodb_search_index_size", batchToCoordinator,
+             batchToPrometheus);
+  auto gaugeToCoordinator = [](ClusterMetricsFeature::Metrics& metrics,
+                               std::string_view name, velocypack::Slice labels,
+                               velocypack::Slice value) {
+    auto labelsStr = labels.stringView();
+    auto end = labelsStr.find(",shard=\"");
+    TRI_ASSERT(end != std::string_view::npos);
+    labelsStr = labelsStr.substr(0, end);
+    metrics::ClusterMetricsFeature::MetricKey key{name, labelsStr};
+    std::get<uint64_t>(metrics[key]) += value.getNumber<uint64_t>();
+  };
+  metric.add("arangodb_search_num_failed_commits", gaugeToCoordinator);
```

Comment:
`gaugeToCoordinator` sounds as if the metrics used here and in the following lines would all be gauges. 
But I think most of them are actually counters, not gauges.
I know it is just a name of an internal lambda, but I found the name "gauge" somewhat misleading here.

Reply:
But I suppose not in this PR (These metrics already exist as gauge)